src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -303,25 +305,26 @@ public boolean isEnabled() {
 	    return true;
     }
     public void initEncoders() {
-
-        referenceAvgPosition = getAveragePosition();
+    	trySetSparkMaxPosition(frontLeftEncoder,0);
+    	trySetSparkMaxPosition(frontLeftEncoder,0);
+    	trySetSparkMaxPosition(rearLeftEncoder,0);
+    	trySetSparkMaxPosition(frontRightEncoder,0);
+    	trySetSparkMaxPosition(rearRightEncoder,0);  
+        referenceAvgPosition = 0.0;
             	
     }
     public void resetEncoders() {
         //This approach is simpler than referencing each individual encoder.
         //normally that's not ok, but in this case, it is safe because we dont expose the individual encoders anywhere
-    	//referenceAvgPosition = getAveragePosition();        
-    	trySetSparkMaxPosition(frontLeftEncoder,0);
-    	trySetSparkMaxPosition(frontLeftEncoder,0);
-    	trySetSparkMaxPosition(rearLeftEncoder,0);
-    	trySetSparkMaxPosition(frontRightEncoder,0);
-    	trySetSparkMaxPosition(rearRightEncoder,0);    	
+    	referenceAvgPosition = getAveragePosition();        
+  	
```

Comment:
We have a subtle bug.  getAveragePosition uses the old value of referenceAvgPosition to calculate the new value of referenceAvgPosition.  Instead referenceAvgPosition should only be the average of the raw encoder values (the internal avgPos in the method below).  I think we need a separate private method getRawEncoderAveragePosition() that returns the avgPos.